### PR TITLE
Optionally log metric deltas across polls in metric_tool.

### PR DESCRIPTION
**Pull request by ewiseblatt**
Keep stats on number of data points collected.

@jtk54

### CODE REVIEW

File: google/stackdriver_monitoring/spectator_client.py [hunk 1]
```diff
@@ -151,12 +160,67 @@ def __init__(self, options):
...
+
+    previous_keys = set(previous_metrics.keys())
+    keys = set(metrics.keys())
+    new_keys = keys.difference(previous_keys)
+    same_keys = keys.intersection(previous_keys)
+    lost_keys = previous_keys.difference(keys)
+    lines = []
+
+    if lost_keys:
+      lines.append('Lost metrics for:\n  - {0}\n'
```
Comment (jtk54) [inline google/stackdriver_monitoring/spectator_client.py]: What does 'Lost metrics' mean? It makes sense where the `lost_keys` come from, but the terminology seems weird.

File: google/stackdriver_monitoring/spectator_client.py [hunk 2]
```diff
@@ -151,12 +160,67 @@ def __init__(self, options):
     self.__host = options['host']
     self.__prototype = None
     self.__default_scan_params = {'tagNameRegex': '.+'}
+    self.__previous_scan_lock = threading.Lock()
+    self.__previous_scan = {} if options.get('log_metric_diff') else None
```
Comment (jtk54) [inline google/stackdriver_monitoring/spectator_client.py]: Does `options.get('log_metric_diff')` throw a `KeyError` if `'log_metric_diff'` isn't present? Make it default to `None` if it does.

File: google/stackdriver_monitoring/spectator_client.py [hunk 3]
```diff
@@ -178,12 +242,31 @@ def collect_metrics(self, host, port, params=None):
     url = 'http://{host}:{port}/spectator/metrics{query}'.format(
         host=host, port=port, query=query)
     response = urllib2.urlopen(url)
+
     all_metrics = json.JSONDecoder(encoding='utf-8').decode(response.read())
+    try:
+      self.__log_scan_diff(host, port, all_metrics.get('metrics', {}))
```
Comment (jtk54) [inline google/stackdriver_monitoring/spectator_client.py]: Seems like control flow here to call this only if `--log_metric_diff` would be clearer than calling this each time and setting internal object state based on the flag.

File: google/stackdriver_monitoring/spectator_client.py (hunk 2 above)
Comment (ewiseblatt) [inline google/stackdriver_monitoring/spectator_client.py]: get doesnt throw key error. It returns None. [] throws key error.

File: google/stackdriver_monitoring/spectator_client.py (hunk 1 above)
Comment (ewiseblatt) [inline google/stackdriver_monitoring/spectator_client.py]: Chanted to Started/Stopped rather than Added/Lost

**Pull request merged**